currency/currencies.go

```diff
@@ -55,6 +55,14 @@ func (c Currencies) Join() string {
 
 // UnmarshalJSON conforms type to the umarshaler interface
 func (c *Currencies) UnmarshalJSON(d []byte) error {
+	if d[0] == '[' {
+		var currs []Code
+		if err := json.Unmarshal(d, &currs); err != nil {
+			return err
+		}
+		*c = currs
+		return nil
```

Comment:
```suggestion
		return json.Unmarshal(d, (*[]Code)(c))
```

You can break recursion without the allocation here

Reply:
You're out of control! I'm surprised at the result
```
// GloriousButts
// BenchmarkButts-10    	  513199	      2344 ns/op	    1448 B/op	      29 allocs/op
// ShazButts
// BenchmarkButts-10    	  568749	      2094 ns/op	    1184 B/op	      24 allocs/op
func BenchmarkButts(b *testing.B) {
	var unmarshalHere Currencies
	j := []byte(`["btc","usd","ltc","bro","things"]`)
	for range b.N {
		_ = json.Unmarshal(j, &unmarshalHere)
	}
}
```